### PR TITLE
Link v3: Assets and component views

### PR DESCRIPTION
**Pull request by davidme-stripe**
## Summary
Bring back a grab bag of assets, colors, and component views for Link. These are all self-contained and don't change the current functionality, but will be used by the ViewControllers in later PRs.

The black and white Link logo in the snapshot tests is a known issue — we'll have updated logo assets before launch.

This code was all reviewed previously, but we should give it another look as we merge it back in.

## Motivation
Link v2

## Testing
Resurrected some snapshot tests

## Changelog
None

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Resources/Localizations/en.lproj/Localizable.strings [hunk 1]
```diff
@@ -70,6 +70,9 @@
 /* Text providing link to terms for ACH payments */
 "By continuing, you agree to authorize payments pursuant to <terms>these terms</terms>." = "By continuing, you agree to authorize payments pursuant to <terms>these terms</terms>.";
 
+/* Mandate text displayed when paying via Link instant debit. */
+"By continuing, you agree to authorize payments pursuant to these <terms>terms</terms>." = "By continuing, you agree to authorize payments pursuant to these <terms>terms</terms>.";
+
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Resources/Localizations/en.lproj/Localizable.strings:75]: probably outside the scope of this PR, but for ACH, it's:
```
<terms>these terms</terms>.
```
and for this one it is:
```
these <terms>terms</terms>.
```

File: StripePaymentSheet/StripePaymentSheet/Resources/StripePaymentSheet.xcassets/Link/icon_link_error.imageset/Contents.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
+      "scale" : "1x"
+    },
+    {
+      "idiom" : "universal",
+      "scale" : "2x"
+    },
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Resources/StripePaymentSheet.xcassets/Link/icon_link_error.imageset/Contents.json:10]: no 1x and 2x assets?

File: StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Components/NavigationBar/LinkNavigationBar.swift [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    }
+
+    override func safeAreaInsetsDidChange() {
+        super.safeAreaInsetsDidChange()
+        invalidateIntrinsicContentSize()
+    }
+
+    private func update() {
+//        TODO(link): Set isLoggedIn once the SDK supports logging in again
+//        let isLoggedIn = linkAccount?.isLoggedIn ?? false
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Components/NavigationBar/LinkNavigationBar.swift:173]: I"m guessing TODO is up to date?

File: StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Components/Toast/LinkToast.swift [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        /// Space between the icon and label.
+        static let spacing: CGFloat = 8
+        static let animationDuration: TimeInterval = 0.2
+        static let animationTravelDistance: CGFloat = 40
+        static let defaultDuration: TimeInterval = 2
+    }
+
+    enum ToastType {
+        case success
+    }
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Components/Toast/LinkToast.swift:29]: This is odd -- an enum with a single case?  I'm fine to leave as is, this doesn't seem needed.

File: StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Components/Toast/LinkToast.swift [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+        #endif
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + duration) {
+            self.hide()
+        }
+    }
+
+    /// Hides toast.
+    ///
+    /// You normally don't need to call this, as the toast will hide on its own after an specific duration.
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Components/Toast/LinkToast.swift:142]: This comment is vague-- it explains that we don't normally need to call this, but doesn't describe when we should be calling it.

File: StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Views/LinkInstantDebitMandateView.swift [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+    private func formattedLegalText() -> NSAttributedString {
+        let string = STPLocalizedString(
+            "By continuing, you agree to authorize payments pursuant to these <terms>terms</terms>.",
+            "Mandate text displayed when paying via Link instant debit."
+        )
+
+        let formattedString = NSMutableAttributedString()
+
+        STPStringUtils.parseRanges(from: string, withTags: Set<String>(links.keys)) { string, matches in
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Views/LinkInstantDebitMandateView.swift:76]: We should be able to leverage:
STPStringUtils.applyLinksToString

At the very least add a todo, if you feel it is out of scope for this PR

File: StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Components/NavigationBar/LinkNavigationBar.swift (hunk 3 above)
Comment (davidme-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Components/NavigationBar/LinkNavigationBar.swift:173]: Yep, we add it in a future PR

File: StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Components/Toast/LinkToast.swift (hunk 4 above)
Comment (davidme-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Components/Toast/LinkToast.swift:29]: That... is weird. I think there should be an error toast too, I'm fine to leave this in for now.

File: StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Components/Toast/LinkToast.swift (hunk 5 above)
Comment (davidme-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Components/Toast/LinkToast.swift:142]: I think it's for if you want to hide it manually. Like maybe you have a "success" toast, then an error immediately occurs and you want to cancel the first toast?

File: StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Views/LinkInstantDebitMandateView.swift (hunk 6 above)
Comment (davidme-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Views/LinkInstantDebitMandateView.swift]: Good call

**Review by wooj-stripe — APPROVED** [review]

**Pull request merged**